pkg/scheduler/framework/runtime/framework.go

```diff
@@ -199,6 +200,12 @@ func WithExtenders(extenders []framework.Extender) Option {
 	}
 }
 
+func WithComponentConfig(cc *config.KubeSchedulerConfiguration) Option {
```

Comment:
Similarly to the ideas above, this could be:

`WithProfileCapturer(func (*config.KubeSchedulerProfile))`

and the code in scheduler.go is in charge of the aggregation.